src/ga4gh/vrs/extras/translator.py

```diff
@@ -9,16 +9,29 @@
 import re
 from abc import ABC
 from collections.abc import Mapping
+from typing import Protocol
 
 from ga4gh.core import ga4gh_identify
 from ga4gh.vrs import models, normalize
-from ga4gh.vrs.dataproxy import _DataProxy
+from ga4gh.vrs.dataproxy import SequenceProxy, _DataProxy
 from ga4gh.vrs.extras.decorators import lazy_property
+from ga4gh.vrs.normalize import denormalize_reference_length_expression
 from ga4gh.vrs.utils.hgvs_tools import HgvsTools
 
 _logger = logging.getLogger(__name__)
 
 
+class VariationToProtocol(Protocol):
+    """Protocol for translating VRS objects to other string expressions
+
+    This protocol defines a callable interface for translating a variation
+    string into a VRS object, with optional keyword arguments for customization.
+    """
+
+    def __call__(self, var: str, **kwargs) -> models._VariationBase | None:
+        """Translate variation `var` to VRS object"""
+
+
```

Comment:
Should it be this?

```suggestion
class VariationToProtocol(Protocol):
    """Protocol for translating VRS objects to other string expressions

    This protocol defines a callable interface for translating a VRS object
    into variation strings, with optional keyword arguments for customization.
    """

    def __call__(self, vo: models._VariationBase, **kwargs) -> list[str]:
        """Translate vrs object `vo` to variation string expressions"""
```

Reply:
We might also want to consider a more descriptive name than `VariationToProtocol`, maybe like `VrsToStrFmtProtocol` or something similar?

Reply:
@korikuzma I like the idea of adding `Str` into the class name. I'll also add a `From` protocol